Arm/rover_arm_system.hpp

```diff
@@ -3,58 +3,199 @@
+    Fingers fingers;
+  };
+
   RoverArmSystem(sjsu::arm::Joint & rotunda,
                  sjsu::arm::Joint & shoulder,
                  sjsu::arm::Joint & elbow,
                  sjsu::arm::WristJoint wrist)
-      : Rotunda(rotunda), Shoulder(shoulder), Elbow(elbow), Wrist(wrist)
+      : rotunda_(rotunda), shoulder_(shoulder), elbow_(elbow), wrist_(wrist)
   {
   }
   void Initialize()
   {
-    Rotunda.Initialize();
-    Shoulder.Initialize();
-    Elbow.Initialize();
-    Wrist.Initialize();
+    rotunda_.Initialize();
+    shoulder_.Initialize();
+    elbow_.Initialize();
+    wrist_.Initialize();
   }
 
   void PrintRoverData(){};
 
-  std::string GETParameters(){};
+  std::string GETParameters()
+  {
+    return "Fill";
+  };
 
-  std::string ParseJSONResponse(){};
+  std::string ParseJSONResponse()
+  {
+    return "Fill";
+  };
 
-  void HandleRoverMovement(){};
+  void HandleArmMovement()
+  {
+    // rotunda
+    units::angular_velocity::revolutions_per_minute_t rotunda_speed(
+        static_cast<float>(mc_data.rotunda_speed));
+    units::angle::degree_t rotunda_angle(
+        static_cast<float>(mc_data.rotunda_angle));
+    // shoulder
+    units::angular_velocity::revolutions_per_minute_t shoulder_speed(
+        static_cast<float>(mc_data.shoulder_speed));
+    units::angle::degree_t shoulder_angle(
+        static_cast<float>(mc_data.shoulder_angle));
+    // elbow
+    units::angular_velocity::revolutions_per_minute_t elbow_speed(
+        static_cast<float>(mc_data.elbow_speed));
+    units::angle::degree_t elbow_angle(static_cast<float>(mc_data.elbow_angle));
+    // wrist
+    units::angular_velocity::revolutions_per_minute_t wrist_speed(
+        static_cast<float>(mc_data.wrist_speed));
+    // units::angle::degree_t
+    // wrist_roll(static_cast<float>(mc_data.wrist_roll)); units::angle::degree_t
+    // wrist_pitch(static_cast<float>(mc_data.wrist_pitch));
+
+    // TODO: implement different arm drive modes in this function.
+
+    // D drive mode
+    rotunda_.SetSpeed(rotunda_speed);
+    rotunda_.SetPosition(rotunda_angle);
+    shoulder_.SetSpeed(shoulder_speed);
+    shoulder_.SetPosition(shoulder_angle);
+    elbow_.SetSpeed(elbow_speed);
+    elbow_.SetPosition(elbow_angle);
+  }
+
+  void HomeArm()
+  {
+    // first getting the accelerometer readings/storing readings
+    Accelerometer::Acceleration_t rotunda_acceleration =
+        rotunda_.GetAccelerometerData();
+    Accelerometer::Acceleration_t shoulder_acceleration =
+        shoulder_.GetAccelerometerData();
+    // then homing the different parts of the arm
+    HomeShoulder(rotunda_acceleration, shoulder_acceleration);
+
+    Accelerometer::Acceleration_t elbow_acceleration =
+        elbow_.GetAccelerometerData();
+    HomeElbow(rotunda_acceleration, elbow_acceleration);
+
+    HomeWrist();
+  }
+  void HomeShoulder(Accelerometer::Acceleration_t rotunda_acceleration,
+                    Accelerometer::Acceleration_t shoulder_acceleration)
```

Comment:
Might also be good idea to have Accelerometer::Acceleration_t member variables or another struct within this class for storing the acceleration values. Would remove the need to pass arguments which can get messy if not careful.